File: 383.赎金信.java
/*
 * @lc app=leetcode.cn id=383 lang=java
 *
 * [383] 赎金信
 */

// @lc code=start
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        short[] count = new short[26];
        // for (char ch : ransomNote.toCharArray())
        //     count[ch-'a']--;
        // for (char ch: magazine.toCharArray())
        //     count[ch-'a']++;
        // for (short s : count) {
        //     if (s<0)
        //         return false;
        // }
        // return true;
        int idx = 0;
        for (char ch: ransomNote.toCharArray()) {
            idx = magazine.indexOf(ch, count[ch-'a']);
            if (idx == -1)
                return false;
            count[ch-'a'] = (short)(idx + 1);
        }
        return true;
    }
}
// @lc code=end

File: 383.赎金信.py
#
# @lc app=leetcode.cn id=383 lang=python3
#
# [383] 赎金信
#

# @lc code=start


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        # 我的
        # li = [0 for i in range(26)]
        # for ch in ransomNote:
        #     li[ord(ch) - 97] -= 1
        # for ch in magazine:
        #     li[ord(ch) - 97] += 1
        # for i in li:
        #     if i < 0:
        #         return False
        # return True
        li = [0] * 26
        for ch in ransomNote:
            idx = magazine.find(ch, li[ord(ch)-97])
            if idx == -1:
                return False
            li[ord(ch) - 97] = idx + 1
        return True

# @lc code=end
